API/projectClass/customFileClass.m

```diff
@@ -153,41 +160,31 @@ function delete(obj)
 
             % Make an 'inputParser' object...
             p = inputParser;
-
-            addParameter(p,'name','', @(x) isText(x))
-            addParameter(p,'filename','', @(x) isText(x))
-            addParameter(p,'functionName','', @(x) isText(x))
-            addParameter(p,'language','', @(x) isText(x) || isenum(x))
-            addParameter(p,'path','', @(x) isText(x)) 
+            addParameter(p, 'name', obj.varTable{row, 1}{:}, @(x) isText(x))
+            addParameter(p, 'filename', obj.varTable{row, 2}{:}, @(x) isText(x))
+            addParameter(p, 'functionName', obj.varTable{row, 3}{:}, @(x) isText(x))
+            addParameter(p, 'language', obj.varTable{row, 4}{:}, @(x) isText(x) || isenum(x))
+            addParameter(p, 'path', obj.varTable{row, 5}{:}, @(x) isText(x)) 
             parse(p, varargin{:});
                 
             results = p.Results;
             
             % Any fields in results that are not empty are being set,
             % so call the relevant set method for these (which will carry
             % out some additional checks)
-            if ~isempty(results.filename)
-                obj.setFileName(row,results.filename);
-            end
-
-            if ~isempty(results.functionName)
-                obj.setFunctionName(row,results.functionName);
-            end
+            obj.setFileName(row, results.filename);
             
-            if ~isempty(results.language)
-                results.language = validateOption(results.language, 'supportedLanguages', obj.invalidLanguageMessage).value;
-                obj.setCustomLanguage(row, results.language);
-            end
+            obj.setFunctionName(row, results.functionName);
             
-            if ~isempty(results.path)
-                % NOT IMPLEMENTED
-                obj.setCustomPath(row,results.path);
-            end
+            obj.setCustomLanguage(row, results.language);
             
-            if ~isempty(results.name)
-                obj.setCustomName(row,results.name);
-            end            
+            if ~strcmp(results.path, obj.varTable{row, 5}{:})
+                obj.setCustomPath(row, results.path);
+            end
 
+            if ~strcmp(results.name, obj.varTable{row, 1}{:})
+                obj.setCustomName(row, results.name);
+            end
```

Comment:
Why do these two fields have if statements checking they have changed whilst the others do not?

Reply:
Because resetting the name led to a duplicate error but I will fix